tensorflow/core/util/mkl_util.h

```diff
@@ -1843,6 +1875,14 @@ class LRUCache {
   // The front of the list contains the key of the most recently accessed
   // entry, while the back of the list is the least recently accessed entry.
   std::list<string> lru_list_;
+
+#ifdef DNNL_AARCH64_USE_ACL
+  // Guards access to the cache and LRU list
+  mutex mu_;
+
+  // The keys that are currently under creation
+  std::set<string> in_flight_;
```

Comment:
I think we should decouple `in_flight_` from `lru_list_`. Let's add another mutex just for it, i.e., `in_flight_mu_`?

Reply:
Added in ecb7614